aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -3798,24 +3812,44 @@ func resourceAviatrixTransitGatewayDelete(d *schema.ResourceData, meta interface
 			// Double the backoff time after each failed try
 			backoff *= 2
 		}
+		if cloudType == goaviatrix.EDGEEQUINIX {
+			err := deleteZtpFile(gateway.GwName, d.Get("vpc_id").(string), d.Get("ztp_file_download_path").(string))
+			if err != nil {
+				return fmt.Errorf("failed to delete ZTP file: %s", err)
+			}
+		}
 	}
 	gateway.GwName = d.Get("gw_name").(string)
 	err := client.DeleteGateway(gateway)
 	if err != nil {
 		return fmt.Errorf("failed to delete Aviatrix Edge Transit Gateway: %s", err)
 	}
+	if cloudType == goaviatrix.EDGEEQUINIX {
+		err := deleteZtpFile(gateway.GwName, d.Get("vpc_id").(string), d.Get("ztp_file_download_path").(string))
+		if err != nil {
+			return fmt.Errorf("failed to delete ZTP file: %s", err)
+		}
+	}
 
 	return nil
 }
 
+func deleteZtpFile(gatewayName, vpcId, ztpFileDownloadPath string) error {
+	fileName := ztpFileDownloadPath + "/" + gatewayName + "-" + vpcId + "-cloud-init.txt"
+	err := os.Remove(fileName)
+	if err != nil {
```

Comment:
Might want to change this to something like
```suggestion
	if err != nil && !errors.Is(err, fs.ErrNotExist) {
```

That way, if the file is already deleted, the function will still return without error.

Reply:
Also, I'd add some unit tests for this function to confirm that the behavior is sane.

Reply:
Still not resolved -- based on earlier comment, you still need to check for `ErrNotExist` if you want this to work as you expect. `os.Remove` will return `ErrNotExist` (`syscall.ENOENT`) if the file does not exist.